src/vs/workbench/contrib/positronNotebook/browser/adapters/PositronNotebookCellViewModel.ts

```diff
@@ -0,0 +1,354 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Emitter } from '../../../../../base/common/event.js';
+import { Disposable } from '../../../../../base/common/lifecycle.js';
+import { Mimes } from '../../../../../base/common/mime.js';
+import { ISettableObservable, observableValue } from '../../../../../base/common/observable.js';
+import { generateUuid } from '../../../../../base/common/uuid.js';
+import { IEditorCommentsOptions } from '../../../../../editor/common/config/editorOptions.js';
+import { IPosition } from '../../../../../editor/common/core/position.js';
+import { Range } from '../../../../../editor/common/core/range.js';
+import { Selection } from '../../../../../editor/common/core/selection.js';
+import { ITextModel, IModelDeltaDecoration } from '../../../../../editor/common/model.js';
+import { IConfigurationService } from '../../../../../platform/configuration/common/configuration.js';
+import { ICellViewModel, ICommonCellViewModelLayoutChangeInfo, INotebookCellDecorationOptions, CellLayoutState, IEditableCellViewModel, CellFocusMode, CodeCellLayoutInfo, ICellOutputViewModel, CellEditState } from '../../../notebook/browser/notebookBrowser.js';
+import { CellViewModelStateChangeEvent, NotebookLayoutInfo } from '../../../notebook/browser/notebookViewEvents.js';
+import { NotebookCellTextModel } from '../../../notebook/common/model/notebookCellTextModel.js';
+import { INotebookCellStatusBarItem } from '../../../notebook/common/notebookCommon.js';
+import { IPositronNotebookInstance } from '../IPositronNotebookInstance.js';
+import { IPositronNotebookCell } from '../PositronNotebookCells/IPositronNotebookCell.js';
+import { PositronCellOutputViewModel } from './PositronCellOutputViewModel.js';
+
+export class PositronNotebookCellViewModel extends Disposable implements ICellViewModel {
```

Comment:
I wonder if we should just refactor the positron notebook cells to fit this rather than wrapping. They are already themselves a wrapper around a cell model.

Reply:
I was wondering about this as well, I think I agree